### DISCUSSION TITLE
Tweaks to work with Nano IoT and FRAM

### DISCUSSION THREAD

**Discussion by violetteta8** [micbay/RaceTimerAndController#27] [Ideas]
Hello, I've been working on this project and it is fantastic! Thank you! I look forward to sharing once I have the final product built. I plan to have it modular between two tracks (1/64th and 1/32).

I initially planned to use a Nano IoT as I have an Arduino Cloud account. The thought would be to make a dashboard that updates as races finish. Then, it could be displayed over the track or on the wall on a tablet or laptop screen (or even cast to a smart TV). This would also allow to store historic results when the system gets turned off. I had not realized the board architecture was different, so the ISR() parts would not compile. I'm not familiar enough with it to really dive in and make the tweaks necessary, so I elected to go with a standard Nano which has worked great so far.

I was also had an idea of adding an I2C FRAM chip to store the top 3 fastest laps with the racer's name. This would allow to pull the data back from the chip up boot up to then have a long-term record of the fastest laps. Then, have a "Reset" to erase the records if the user wanted to.

Thanks again for your efforts and thorough write up! So much fun!

**Comment by micbay** [comment]
Hi, it's great to hear you're having fun with the project.

Yeah, the type of pin change interrupt feature used, is a hardware level feature of the ATmega processor that is deeper than just the signal handling pinout. I haven't worked with the [Nano 33 IoT](https://docs.arduino.cc/resources/datasheets/ABX00027-datasheet.pdf), but I had a quick look at the datasheet and it uses a Cortex M0+ SAMD21 processor instead of an ATmega328 or similar ATmega based processor used on the standard Nano or Uno. Doesn't look like the SAMD21 has an equivalent hardware interrupt that can be easily swapped in.

It would be nice to add a data out, ideally in real time, to support output to a computer or view display. In this [previous project discussion](https://github.com/micbay/RaceTimerAndController/discussions/23), I helped another builder work out adding serial communication so they could use a PC to start a race, and dump race data to.

When i get a chance I've wanted to make a fork or version that removes the Adafruit startlight to get that memory back, and add serial data out and race control. Perhaps an approach that could achieve what you are looking for from the IoT Nano could be done by using an add on transmitter, or transceiver (transmitter and receiver) board like the ESP32 that takes that serial and sends it via bluetooth or wifi to whatever IoT device.

I look forward to seeing your setup when you're finished.
Cheers
